source/patterns/@aws-solutions-constructs/aws-sqs-pipes-stepfunctions/lib/index.ts

```diff
@@ -0,0 +1,185 @@
+ *  and limitations under the License.
+ */
+
+import * as defaults from '@aws-solutions-constructs/core';
+import * as sfn from 'aws-cdk-lib/aws-stepfunctions';
+import * as logs from 'aws-cdk-lib/aws-logs';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as cloudwatch from 'aws-cdk-lib/aws-cloudwatch';
+import * as sqs from 'aws-cdk-lib/aws-sqs';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import * as pipes from 'aws-cdk-lib/aws-pipes';
+import * as kms from 'aws-cdk-lib/aws-kms';
+import { PipesLogLevel } from '@aws-solutions-constructs/core';
+export { PipesLogLevel } from '@aws-solutions-constructs/core';
+// Note: To ensure CDKv2 compatibility, keep the import statement for Construct separate
+import { Construct } from 'constructs';
+
+/**
+ * @summary The properties for the SnsToSqs class.
+ */
+export interface SqsToPipesToStepfunctionsProps {
+
+  // *******************
+  // Queue Props
+  // *******************
+
+  /**
+   * An optional, existing SQS queue to be used instead of the default queue. Providing both this and queueProps will cause an error.
+   */
+  readonly existingQueueObj?: sqs.Queue,
+  /**
+   * Optional user provided properties to override the default properties for the SQS queue.
+   */
+  readonly queueProps?: sqs.QueueProps,
+  /**
+   * Whether to encrypt the Queue with a customer managed KMS key (CMK). This is the default
+   * behavior, and this property defaults to true - if it is explicitly set to false then the Queue
+   * is encrypted with an Amazon managed KMS key. For a completely unencrypted Queue (not recommended),
+   * create the Queue separately from the construct and pass it in using the existingQueueObject. Since
+   * SNS subscriptions do not currently support SQS queues with AWS managed encryption keys, setting this
+   * to false will always result in an error from the underlying CDK - we have still included this property
+   * for consistency with topics and to be ready if the services one day support this functionality.
+   */
+  readonly encryptQueueWithCmk?: boolean,
+  /**
+   * An optional subset of key properties to override the default properties used by constructs (enableKeyRotation: true).
+   * These properties will be used in constructing the CMK used to encrypt the SQS queue.
+   */
+  readonly queueEncryptionKeyProps?: kms.KeyProps,
+  /**
+   * 	An optional CMK that will be used by the construct to encrypt the new SQS queue.
+   */
+  readonly existingQueueEncryptionKey?: kms.Key,
+  /**
+   * Whether to create a secondary queue to be used as a dead letter queue.
+   *
+   * default = true.
+   */
+  readonly deployDeadLetterQueue?: boolean,
+  /**
+   * Optional user-provided props to override the default props for the dead letter SQS queue.
+   */
+  readonly deadLetterQueueProps?: sqs.QueueProps,
+  /**
+   * The number of times a message can be unsuccessfully dequeued before being moved to the dead letter
+   * queue. Defaults to 15.
```

Comment:
To me, 15 retries seems high for a default. Is this number coming from a best practice doc some where?

Reply:
It does seem high, it was set as a default 4 years ago. But changing it at this point would alter running workloads so I think just documenting it is our best option.